src/prefect/engine.py

```diff
@@ -1861,14 +1865,28 @@ def resolve_input(expr, context):
 
         return result_by_state.get(state)
 
-    return visit_collection(
-        parameters,
-        visit_fn=resolve_input,
-        return_data=return_data,
-        max_depth=max_depth,
-        remove_annotations=True,
-        context={},
-    )
+    resolved_parameters = {}
+    for parameter, value in parameters.items():
+        try:
+            resolved_parameters[parameter] = visit_collection(
+                value,
+                visit_fn=resolve_input,
+                return_data=return_data,
+                # we're manually going 1 layer deeper here
+                max_depth=max_depth - 1,
```

Comment:
@madkinsz 

the remaining errors seem to be caused by:
https://github.com/PrefectHQ/prefect/blob/main/src/prefect/engine.py#L1027
```
    parameters = await resolve_inputs(parameters, max_depth=1)
```
This will will lead errors due the `max_depth`. I'm having trouble reasoning if the `max_depth` should change in that line OR changed here because we're manually going 1 layer deep into the parameters?

Reply:
I may be.... outside of my depth...